workflow-examples/src/test/java/com/redhat/parodos/notification/integration/NotificationRecordTest.java

```diff
@@ -0,0 +1,135 @@
+		}
+		catch (ApiException e) {
+			log.error("Exception when calling NotificationRecordApi#update:", e);
+			fail("Exception when calling NotificationRecordApi#update");
+		}
+
+		validateNotificationRecordCount(testName, 2);
+		list = validateNotificationRecordList(testName, 3);
+
+		notificationRecord = list.getContent().get(1);
+
+		try {
+			this.recordApiInstance.deleteNotification(notificationRecord.getId());
+		}
+		catch (ApiException e) {
+			log.error("Exception when calling NotificationRecordApi#delete:", e);
+			fail("Exception when calling NotificationRecordApi#delete");
+		}
+
+		validateNotificationRecordCount(testName, 1);
+		list = validateNotificationRecordList(testName, 2);
+
+	}
+
+	private void validateNotificationRecordCount(String testName, int notificationRecordsExpectedCount) {
+		logTestStep(testName, "Count Notification Records for the user");
+		try {
+			Integer count = this.recordApiInstance.countUnreadNotifications("UNREAD");
+			log.info("Found ", count, "notification records for the user");
+			assertEquals(notificationRecordsExpectedCount, count.intValue());
+		}
+		catch (ApiException e) {
+			log.error("Exception when calling NotificationRecordApi#count:", e);
+			fail("Exception when calling NotificationRecordApi#count");
+		}
+	}
+
+	private PageNotificationRecordResponseDTO validateNotificationRecordList(String testName,
+			int notificationRecordsExpectedCount) {
+		logTestStep(testName, "List Notification Records for the user");
+		try {
+			PageNotificationRecordResponseDTO result = this.recordApiInstance.getNotifications(new Pageable(), null,
+					null);
+			List<NotificationRecordResponseDTO> content = result.getContent();
+			log.info(content.toString());
+			assertEquals(notificationRecordsExpectedCount, content.size());
+			return result;
+		}
+		catch (ApiException e) {
+			log.error("Exception when calling NotificationRecordApi#get:", e);
+			fail("Exception when calling NotificationRecordApi#get");
+		}
+		return null;
+	}
+
+	private void createNotificationMessage(String messageType, String body, String subject, String username) {
+		NotificationMessageCreateRequestDTO notificationMessageCreateRequestDTO = new NotificationMessageCreateRequestDTO();
+
+		notificationMessageCreateRequestDTO.messageType(messageType);
+		notificationMessageCreateRequestDTO.body(body);
+		notificationMessageCreateRequestDTO.subject(subject);
+		if (username != null) {
+			notificationMessageCreateRequestDTO.usernames(Arrays.asList(username));
+		}
+
+		try {
+			this.messageApiInstance.create(notificationMessageCreateRequestDTO);
+		}
+		catch (ApiException e) {
+			log.error("Exception when calling NotificationMessageApi#create:", e);
+			fail("Exception when calling NotificationMessageApi#create:");
+		}
+	}
+
+	private void logTestStep(String testName, String stepDescription) {
+		log.info("############## " + testName + ": " + stepDescription);
```

Comment:
Is this really needed? if we need to log in a test maybe can be splitted in two?

Reply:
at the moment you need to add logs inside the test, means that the log may have too many complexity and maybe you can split in more.

Reply:
This is integration test, each test may need to validate all operations:
1. Create notification
2. Delete notification
3. get notification
4. count notifications
Therefore, I prefer to have sub function to each operation and than each test function can call to the sub functions, and I created the logs this way to make the flow more clear.
I agree that this is shouldn't be done in unit tests, but this is not the case in this integration test